mlflow/types/chat.py

```diff
@@ -80,9 +80,20 @@ class RequestMessage(BaseModel):
 
     role: str
     content: Optional[ContentType] = None
+    # NB: In the actual OpenAI chat completion API spec, these fields only
+    #   present in either the request or response message (tool_call_id is only in
+    #   the request, while the other two are only in the response).
+    #   Strictly speaking, we should separate the request and response message types
+    #   to match OpenAI's API spec. However, we don't want to do that because we the
+    #   request and response message types are not distinguished in many parts of the
+    #   codebase, and also we don't want to ask users to use two different classes.
+    #   Therefore, we include all fields in this class, while marking them as optional.
+    # TODO: Define a sub classes for different type of messages (request/response, and
+    #   system/user/assistant/tool, etc), and create a factory function to allow users
+    #   to create them without worrying about the details.
     tool_calls: Optional[list[ToolCall]] = Field(None, min_items=1)
-    tool_call_id: Optional[str] = None
     refusal: Optional[str] = None
+    tool_call_id: Optional[str] = None
 
 
 class ParamType(BaseModel):
```

Comment:
unrelated but after playing around a bit, i realize `type` is optional. for example, this is a valid tool definition for openai SDK:

```
{
    "type": "function",
    "function": {
        "name": "add_numbers",
        "parameters": {
            "type": "object",
            "properties": {
                "a": {},
                "b": {},
            },
        },
    },
}
```


could we fix the below line to be

```
type: Optional[Literal["string", "number", "integer", "object", "array", "boolean", "null"]] = None
```